### PR TITLE
Bugfixes and Issue fixes

### LINKED ISSUE #11: Formula evaluation happens multiple times
Hi @fhm84,

I use a random number and a reference in my test.

Cell A1: =ZUFALLSBEREICH(1;999)
Cell A2: =A1

In my Excel-Sheet both cells A1 and A2 are the same. But jexunit evaluates these formular multiple times. Do you have any suggestion for that?

### LINKED ISSUE #13: Format of numbers is getting ignored
Hi @fhm84,

I have a command which accepts a HashMap. In this map date and number formats are getting ignored. Following command fails for a number "1" in the excel sheet.

``` 
public class LoadMapCommand {
	@TestCommand("loadmap")
	public void runCommand(HashMap<String, String> map) {
                // fails because "1.0" does not equal "1"
		assertThat(map.get("number"), equalTo("1"));
	}
}
``` 

I have the same problem for date formats, but I use `jexunit.datePattern` config as a workaround.

Regards, Rokko

### PR DESCRIPTION
**Pull request by Rokko11**
- guthub actions
- fix #11 
- fix #13

### CODE REVIEW

File: core/src/main/java/com/jexunit/core/spi/AfterSheet.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.jexunit.core.spi;
+
```
Comment (fhm84) [inline core/src/main/java/com/jexunit/core/spi/AfterSheet.java:2]: Could you please add some explaining documentation?

File: core/src/main/java/com/jexunit/core/spi/BeforeSheet.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.jexunit.core.spi;
+
```
Comment (fhm84) [inline core/src/main/java/com/jexunit/core/spi/BeforeSheet.java:2]: Could you please add some explaining documentation?

File: examples-complex/src/test/java/com/jexunit/examples/businesstests/commands/LoadAndCalculateTestCommand.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 	 *            the TestContext to add the loaded entity to
 	 */
 	@TestCommand
-	public void load(@TestParam long id, TestContext context) {
+	public void load(@TestParam("id") long id, TestContext context) {
```
Comment (fhm84) [inline examples-complex/src/test/java/com/jexunit/examples/businesstests/commands/LoadAndCalculateTestCommand.java:30]: There was no value set for the @TestParam-annotation to test resolving the parameter name!
But this is only working by setting the compiler argument "-parameter" like it was set before in the compiler-plugin configuration!

File: pom.xml [hunk 4]
```diff
@@ -52,25 +52,12 @@ Developers design a "API" for the department so these specialists can write func
...
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-compiler-plugin</artifactId>
-				<version>3.8.1</version>
-				<configuration>
-					<release>8</release>
-					<showDeprecation>true</showDeprecation>
-					<showWarnings>true</showWarnings>
-					<fork>true</fork>
-					<compilerArgs>
-						<arg>-parameters</arg>
```
Comment (fhm84) [inline pom.xml:69]: This is necessary to get/access/resolve the parameter names!

File: core/src/main/java/com/jexunit/core/JExUnitBase.java [hunk 5]
```diff
@@ -211,13 +213,28 @@ public void runCommand(final TestCase<?> testCase) throws Exception {
     }
 
     @BeforeClass
-    public static void initialseTest() {
-        // TODO: add callback?!
+    public static void initialseTest() throws Exception {
+        String clazzName = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.BEFORE_EXCEL);
```
[the remark this answers is not in the document]
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/JExUnitBase.java:217]: Yes. Imagine there is a database-cleanup in a after hook which every developer can activate/deactivate if she wants.

File: core/src/main/java/com/jexunit/core/JExUnitBase.java [hunk 6]
```diff
@@ -211,13 +213,28 @@ public void runCommand(final TestCase<?> testCase) throws Exception {
...
+                instance.run();
+            }
+        }
     }
 
     @AfterClass
-    public static void finalizeTest() {
-        // TODO: add callback?!
+    public static void finalizeTest() throws Exception{
+        String clazzName = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.AFTER_EXCEL);
```
[the remark this answers is not in the document]
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/JExUnitBase.java:229]: same same

File: core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java [hunk 7]
```diff
@@ -123,8 +122,13 @@ private String calculateCommandName(final Class<?> type) {
         String name = type.getSimpleName();
         final String prefix = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.COMMAND_CLASS_PREFIX);
         final String postfix = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.COMMAND_CLASS_POSTFIX);
-        name = StringUtils.removeStartIgnoreCase(name, prefix);
-        name = StringUtils.removeEndIgnoreCase(name, postfix);
+        if (name.toLowerCase().startsWith(prefix.toLowerCase())) {
```
[the remark this answers is not in the document]
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java:125]: I know, but if you want to use JExUnit in a (e.g. maintanence-)project with an apache commons dependency < 2.4 then these lines prevent the usage of JExUnit.

File: core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java [hunk 8]
```diff
@@ -139,8 +143,13 @@ private String calculateCommandName(final Method m) {
         String name = m.getName();
         final String prefix = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.COMMAND_METHOD_PREFIX);
         final String postfix = JExUnitConfig.getStringProperty(JExUnitConfig.ConfigKey.COMMAND_METHOD_POSTFIX);
-        name = StringUtils.removeStartIgnoreCase(name, prefix);
-        name = StringUtils.removeEndIgnoreCase(name, postfix);
+        if (name.toLowerCase().startsWith(prefix.toLowerCase())) {
```
[the remark this answers is not in the document]
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java:146]: same same

File: examples-complex/src/test/java/com/jexunit/examples/businesstests/commands/LoadAndCalculateTestCommand.java (hunk 3 above)
Comment (Rokko11) [reply examples-complex/src/test/java/com/jexunit/examples/businesstests/commands/LoadAndCalculateTestCommand.java:30]: sorry, that's right.

File: core/src/main/java/com/jexunit/core/spi/BeforeSheet.java (hunk 2 above)
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/spi/BeforeSheet.java:2]: I'll fix that.

File: core/src/main/java/com/jexunit/core/spi/AfterSheet.java (hunk 1 above)
Comment (Rokko11) [reply core/src/main/java/com/jexunit/core/spi/AfterSheet.java:2]: I'll fix that.

File: pom.xml (hunk 4 above)
Comment (Rokko11) [reply pom.xml:69]: sorry, that's right.

File: core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (fhm84) [reply core/src/main/java/com/jexunit/core/commands/TestCommandScanner.java:125]: It's absolutely fine, and I think it's a good idea to remove some dependencies ...
But what I mentioned here is the now missing null check!

**Pull request merged**